journalbeat/reader/unit.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2017 Marcus Heese
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code based on logic from journalctl unit filter. i.e. journalctl -u in
+// the systemd source code.
+// See: https://github.com/systemd/systemd/blob/master/src/journal/journalctl.c#L1410
+// and https://github.com/systemd/systemd/blob/master/src/basic/unit-name.c
```

Comment:
@kvch I've looked at the `make check`'s error, this is because the header doesn't match any format that we enforce, ie: the Elasticsearch Apache 2.0 or Elasticsearch's own license.

The files _unit.go_ and _kernel.go_ appear to be a copy of the original files from the other Journalbeat project. IIRC APL2.0 allows this if we keep the header/attribution.

Concerning how to fix the error on our side I see a few options:

1. We actually vendor the files in the _vendor_ directory and use them there.
2. We ask Marcus to make a PR on our repo with our header / give us the rights.
3. We allow exceptions of license on our repo.
4. We recode it.

I am _no lawyer_ and I would like to hear what @urso or @tsg thinks.

**Note:** I haven't looked a the difference between theses files and the original

Reply:
1 or 4 seems what we should do.